### PR TITLE
ensure citations are read with explicit encoding

### LINKED ISSUE #416: non-ASCII input in a CITATION file without a declared encoding
Because of the accent in my family name in [the CITATION file](https://github.com/privefl/bigsnpr/blob/master/inst/CITATION) of [my package](https://github.com/privefl/bigsnpr), `pkgdown::build_site()` fails with error
> Error in tools:::.parse_CITATION_file(file, meta$Encoding) : 
  non-ASCII input in a CITATION file without a declared encoding

I've tried with `pkgdown::build_site(encoding = "UTF-8")`, but it fails also with the same error.
Note that I have also `Encoding: UTF-8` in my DESCRIPTION file.

***
Session info:
```
Session info ---------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.1 (2017-06-30)
 system   x86_64, linux-gnu           
 ui       RStudio (1.0.143)           
 language (EN)                        
 collate  fr_FR.UTF-8                 
 tz       Europe/Paris                
 date     2017-10-05                  

Packages -------------------------------------------------------------------------------
 package    * version    date       source                          
 assertthat   0.2.0      2017-04-11 CRAN (R 3.4.1)                  
 backports    1.1.0      2017-05-22 CRAN (R 3.4.1)                  
 base       * 3.4.1      2017-06-30 local                           
 bigsnpr    * 0.2.1      2017-10-05 local                           
 bigstatsr  * 0.2.3      2017-10-05 local                           
 codetools    0.2-15     2016-10-05 CRAN (R 3.4.1)                  
 colorspace   1.3-2      2016-12-14 CRAN (R 3.4.1)                  
 commonmark   1.4        2017-09-01 CRAN (R 3.4.1)                  
 compiler     3.4.1      2017-06-30 local                           
 crayon       1.3.4      2017-09-16 CRAN (R 3.4.1)                  
 curl         2.8.1      2017-07-21 CRAN (R 3.4.1)                  
 data.table   1.10.4     2017-02-01 cran (@1.10.4)                  
 datasets   * 3.4.1      2017-06-30 local                           
 desc         1.1.1      2017-08-03 CRAN (R 3.4.1)                  
 devtools     1.13.3     2017-08-02 CRAN (R 3.4.1)                  
 digest       0.6.12     2017-01-27 CRAN (R 3.4.1)                  
 evaluate     0.10.1     2017-06-24 CRAN (R 3.4.1)                  
 foreach      1.4.3      2015-10-13 cran (@1.4.3)                   
 ggplot2    * 2.2.1      2016-12-30 CRAN (R 3.4.1)                  
 graphics   * 3.4.1      2017-06-30 local                           
 grDevices  * 3.4.1      2017-06-30 local                           
 grid         3.4.1      2017-06-30 local                           
 gtable       0.2.0      2016-02-26 CRAN (R 3.4.1)                  
 hexbin     * 1.27.1     2016-12-05 CRAN (R 3.4.1)                  
 htmltools    0.3.6      2017-04-28 CRAN (R 3.4.1)                  
 httpuv       1.3.5      2017-07-04 CRAN (R 3.4.1)                  
 httr         1.3.1      2017-08-20 CRAN (R 3.4.1)                  
 iterators    1.0.8      2015-10-13 cran (@1.0.8)                   
 knitr        1.17       2017-08-10 CRAN (R 3.4.1)                  
 lattice      0.20-35    2017-03-25 CRAN (R 3.4.1)                  
 lazyeval     0.2.0      2016-06-12 CRAN (R 3.4.1)                  
 magrittr     1.5        2014-11-22 CRAN (R 3.4.1)                  
 MASS         7.3-47     2017-02-26 CRAN (R 3.4.1)                  
 Matrix       1.2-10     2017-05-03 CRAN (R 3.4.1)                  
 memoise      1.1.0      2017-04-21 CRAN (R 3.4.1)                  
 methods    * 3.4.1      2017-06-30 local                           
 mime         0.5        2016-07-07 CRAN (R 3.4.1)                  
 munsell      0.4.3      2016-02-13 CRAN (R 3.4.1)                  
 parallel     3.4.1      2017-06-30 local                           
 pkgdown      0.1.0.9000 2017-10-05 Github (hadley/pkgdown@c8726f9) 
 pkgload      0.0.0.9000 2017-10-05 Github (r-lib/pkgload@7fd5f55)  
 plyr         1.8.4      2016-06-08 CRAN (R 3.4.1)                  
 purrr        0.2.3      2017-08-02 CRAN (R 3.4.1)                  
 R6           2.2.2      2017-06-17 CRAN (R 3.4.1)                  
 Rcpp         0.12.12    2017-07-15 CRAN (R 3.4.1)                  
 rlang        0.1.2.9000 2017-10-05 Github (tidyverse/rlang@c6b038f)
 rmarkdown    1.6        2017-06-15 CRAN (R 3.4.1)                  
 roxygen2     6.0.1      2017-02-06 CRAN (R 3.4.1)                  
 rprojroot    1.2        2017-01-16 CRAN (R 3.4.1)                  
 rstudioapi   0.7        2017-09-07 CRAN (R 3.4.1)                  
 scales       0.5.0      2017-08-24 CRAN (R 3.4.1)                  
 shiny      * 1.0.5      2017-08-23 CRAN (R 3.4.1)                  
 stats      * 3.4.1      2017-06-30 local                           
 stringi      1.1.5      2017-04-07 CRAN (R 3.4.1)                  
 stringr      1.2.0      2017-02-18 CRAN (R 3.4.1)                  
 tibble       1.3.4      2017-08-22 CRAN (R 3.4.1)                  
 tools        3.4.1      2017-06-30 local                           
 usethis      0.1.0      2017-10-05 Github (r-lib/usethis@8a00ce7)  
 utils      * 3.4.1      2017-06-30 local                           
 withr        2.0.0      2017-07-28 CRAN (R 3.4.1)                  
 xgboost      0.6-4      2017-01-05 cran (@0.6-4)                   
 xml2         1.1.1      2017-01-24 CRAN (R 3.4.1)                  
 xtable       1.8-2      2016-02-05 CRAN (R 3.4.1)                  
 yaml         2.1.14     2016-11-12 CRAN (R 3.4.1)   
```

### PR DESCRIPTION
**Pull request by jayhesselberth**
- fixes #416

### CODE REVIEW

**Comment by pat-s** [comment]
Works for me but needs to be rebased with master to include the most recent pandoc 2.0 fix :)

File: R/build-home-citation.R [hunk 1]
```diff
@@ -24,22 +27,25 @@ data_home_sidebar_citation <- function(pkg = ".") {
   list_with_heading(citation, "Citation")
 }
 
-data_citations <- function(pkg = ".") {
+data_citations <- function(pkg = ".", encoding = "UTF-8") {
   pkg <- as_pkgdown(pkg)
-  cit <- read_citation(pkg$path)
+  cit <- read_citation(pkg$path, encoding)
```
Comment (WastlM) [inline R/build-home-citation.R]: Writing R Extensions says:

> The CITATION file is parsed as R code (in the package’s declared encoding, or in ASCII if none is declared)

(see https://cran.r-project.org/doc/manuals/r-release/R-exts.html#CITATION-files)

So shouldn't `data_citations()` just call `read_citation()` with the encoding derived from `pkg$desc`?

File: R/build-home-citation.R [hunk 2]
```diff
@@ -3,12 +3,15 @@ has_citation <- function(path = ".") {
   file.exists(file.path(path, 'inst/CITATION'))
 }
 
-read_citation <- function(path = ".") {
+read_citation <- function(path = ".", encoding) {
```
Comment (hadley) [inline R/build-home-citation.R]: Default to UTF-8 here too?

File: R/build-home-citation.R (hunk 1 above)
Comment (hadley) [reply R/build-home-citation.R]: I don't really want to get into supporting (and hence testing for) any arbitrary encoding. I think UTF-8 is adequate

**Comment by jayhesselberth** [comment]
This is ready except I mistakenly added e1a3c2c to this branch instead of a local branch I had try to address #493, which is a whole different kettle of fish.

I can remove that commit to merge this, or can continue to work on a more robust approach to fix this and #493.

**Comment by hadley** [comment]
I think we can resolve both problems with a single fix - generate the complete `meta` from the DESCRITPION

**Pull request closed without merging**